addons/knowledge/static/src/js/knowledge_controller.js

```diff
@@ -106,6 +107,29 @@ const KnowledgeArticleFormController = FormController.extend({
         await this._rename(id, event.currentTarget.value);
     },
 
+    /**
+     * When the user clicks on the name of the article, checks if the article
+     * name hasn't been set yet. If it hasn't, it will look for a title in the
+     * body of the article and set it as the name of the article.
+     * @param {Event} event
+     */
+    _onSetArticleName: async function (event) {
```

Comment:
Actually this whole code should be an onchange.

Reply:
A python onchange. As we are in a form view we should not have custom code for something that is basically an onchange.

Reply:
We don't want to have custom code to replace standard features. So no, it should be done by an onchange correctly managed by the form view.